mmdet/models/roi_heads/sparse_roi_head.py

```diff
@@ -0,0 +1,313 @@
+                                    self.stage_loss_weights[stage]
+            object_feats = bbox_results['object_feats']
+
+        return all_stage_loss
+
+    def simple_test(self,
+                    x,
+                    proposal_boxes,
+                    proposal_feats,
+                    img_metas,
+                    imgs_whwh,
+                    rescale=False):
+        """Test without augmentation.
+
+        Args:
+            x (list[Tensor]): list of multi-level img features.
+            proposal_boxes (Tensor): Decoded proposal bboxes, has shape
+                (batch_size, num_proposals, 4)
+            proposal_feats (Tensor): Expanded proposal
+                features, has shape
+                (batch_size, num_proposals, proposal_feature_channel)
+            img_metas (dict): meta information of images.
+            imgs_whwh (Tensor): Tensor with shape (batch_size, 4),
+                    the dimension means
+                    [img_width,img_height, img_width, img_height].
+            rescale (bool): If True, return boxes in original image
+                space. Defaults to False.
+
+        Returns:
+            bbox_results (list[tuple[np.ndarray]]): \
+                [[cls1_det, cls2_det, ...], ...]. \
+                The outer list indicates images, and the inner \
+                list indicates per-class detected bboxes. The \
+                np.ndarray has shape (num_det, 5) and the last \
+                dimension 5 represents (x1, y1, x2, y2, score).
+        """
+        assert self.with_bbox, 'Bbox head must be implemented.'
+        # Decode initial proposals
+        num_imgs = len(img_metas)
+        num_proposals = proposal_boxes.size(1)
+        proposal_list = [proposal_boxes[i] for i in range(num_imgs)]
+        object_feats = proposal_feats
+        for stage in range(self.num_stages):
+            rois = bbox2roi(proposal_list)
+            bbox_results = self._bbox_forward(stage, x, rois, object_feats,
+                                              img_metas)
+            object_feats = bbox_results['object_feats']
+            cls_score = bbox_results['cls_score']
+            proposal_list = bbox_results['detach_proposal_list']
+
+        num_classes = self.bbox_head[-1].num_classes
+        det_bboxes = []
+        det_labels = []
+
+        if self.bbox_head[-1].loss_cls.use_sigmoid:
+            cls_score = cls_score.sigmoid()
+        else:
+            cls_score = cls_score.softmax(-1)[..., :-1]
+        labels = torch.arange(num_classes)[None].repeat(
+            num_proposals, 1).flatten(0, 1).type_as(proposal_feats)
+        for img_id in range(num_imgs):
+            cls_score_per_img = cls_score[img_id]
+            scores_per_img, topk_indices = cls_score_per_img.flatten(
+                0, 1).topk(
+                    self.test_cfg.max_per_img, sorted=False)
+            labels_per_img = labels[topk_indices]
```

Comment:
logics here can be simplified by:
```
bbox_indices = topk_indices // num_classes
labels_per_img = topk_indices % num_classes
bbox_pred_per_img = proposal_list[img_id][bbox_indices]
```
In this way, we do not need to repeat the labels and bbox_pred before slicing.